backend/api/messages.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any, Dict, List, Tuple, cast
+
+from bson import ObjectId
+from flask import Blueprint, jsonify, request
+from flask_jwt_extended import get_jwt_identity, jwt_required
+
+from backend.database_handler_entity import mongo
+
+message_api = Blueprint('message_api', __name__)
+
+
+@message_api.route('/users/<document_id>', methods=["GET"])
+@jwt_required()
+def get_users_with_permissions(document_id: str) -> Tuple[Any, int]:
+    user_id = ObjectId(get_jwt_identity())
+
+    document = mongo.find_document(document_id)
+    if document is None:
+        return jsonify({"message": "No such document"}), 404
+
+    users = mongo.select_company_users(document["company"])
+    users.extend(mongo.get_users_with_permissions(document["_id"]))
+    users = list(filter(lambda x: x["_id"] != user_id, users))
+
+    for user in users:
+        user["_id"] = str(user["_id"])
+
+    return jsonify(users), 200
+
+
+@message_api.route('/message', methods=["POST"])
+@jwt_required()
+def create_message() -> Tuple[Any, int]:
+    user_id = ObjectId(get_jwt_identity())
+    body = request.get_json()
+
+    if not body.get("to_users", False) or not body.get("message", False):
+        return jsonify({"message": "Invalid body content"}), 403
+
+    if isinstance(body["to_users"], str):
```

Comment:
If `to_users` might be a list then let `to_users` will be always a list. So it will simplify your code and will remove this using of `isinstance` that is not really good (actually it is bad from the point of code structure)